Robustness task

rimashahbazyan opened the pull request.

rimashahbazyan committed:
added Docker image
robustness same option eval
prompt robustness eval

rimashahbazyan committed:
added Docker image
robustness same option eval
prompt robustness eval

rimashahbazyan committed:
prompt loading updated

rimashahbazyan committed:
prompt loading updated

rimashahbazyan committed:
yaml based implementations of options and prompt robustness
+ consistency rate implementation

rimashahbazyan committed:
yaml based implementations of options and prompt robustness
+ consistency rate implementation

rimashahbazyan committed:
fixing default generate_until (until) parameter to be eos of tokenizer

rimashahbazyan committed:
fixing default generate_until (until) parameter to be eos of tokenizer

rimashahbazyan committed:
minor refactorings

rimashahbazyan committed:
minor refactorings

rimashahbazyan committed:
added new templates

rimashahbazyan committed:
added new templates

rimashahbazyan committed:
v_03 templates

rimashahbazyan committed:
v_03 templates

rimashahbazyan committed:
macro accuracy for prompt robustness added

rimashahbazyan committed:
macro accuracy for prompt robustness added

rimashahbazyan committed:
option format robustness added

rimashahbazyan committed:
option format robustness added

rimashahbazyan committed:
Options order robustness fixed

rimashahbazyan committed:
Options order robustness fixed

rimashahbazyan committed:
Option format robustness fixed

rimashahbazyan committed:
Option format robustness fixed

rimashahbazyan committed:
metric names adjusted

rimashahbazyan committed:
metric names adjusted

rimashahbazyan committed:
per_option_macro_accuracy fix

rimashahbazyan committed:
per_option_macro_accuracy fix

rimashahbazyan committed:
agi_eval added

rimashahbazyan committed:
agi_eval added

rimashahbazyan committed:
Filtering sampler added

rimashahbazyan committed:
Filtering sampler added

rimashahbazyan committed:
fewshot prompt robustness adedd and minor changes to other mmlu tasks

rimashahbazyan committed:
fewshot prompt robustness adedd and minor changes to other mmlu tasks

rimashahbazyan committed:
lsat_rc fix

rimashahbazyan committed:
lsat_rc fix

rimashahbazyan committed:
agieval_sat_math added

rimashahbazyan committed:
agieval_sat_math added

rimashahbazyan committed:
Filtering sampler refactored

rimashahbazyan committed:
Filtering sampler refactored

rimashahbazyan committed:
Minor refactornig and code dublicate removal
Added robustness_agieval and robustness_mmlu_pro groups

rimashahbazyan committed:
Minor refactornig and code dublicate removal
Added robustness_agieval and robustness_mmlu_pro groups

rimashahbazyan committed:
Readme added

rimashahbazyan committed:
Readme added

rimashahbazyan committed:
Merge branch 'robustness_task' of github.com:rimashahbazyan/lm-evaluation-harness into robustness_task

rimashahbazyan committed:
Merge branch 'EleutherAI:main' into robustness_task

ekmb commented on lm_eval/api/samplers.py:
```diff
@@ -184,9 +183,123 @@ def sample(self, n) -> None:
 
 
+class FilteringSampler(ContextSampler):
+    def __init__(self, docs, task, fewshot_indices=None, rnd=None, filters=[]) -> None:
```
add docstrings to the class and all the functions

rimashahbazyan replied:
done

ekmb commented on lm_eval/api/samplers.py:
```diff
@@ -184,9 +183,123 @@ def sample(self, n) -> None:
+        fewshotex = self.sample(n_samples, filter_dict)
+
+        # get rid of the doc that's the one we're evaluating, if it's in the fewshot
+        # TODO: should we just stop people from using fewshot from same split as evaluating?
```
is this comment still needed?

rimashahbazyan replied:
This was copied from their ContextSampler, I guess we'd better remove it, but I'm not sure

ekmb commented on lm_eval/api/samplers.py:
```diff
@@ -184,9 +183,123 @@ def sample(self, n) -> None:
+         super().__init__(docs, task, fewshot_indices, rnd)
+         self.filters = filters
+    
+    def get_context(self, doc, num_fewshot):
```
docstring

rimashahbazyan replied:
done

ekmb commented on lm_eval/api/samplers.py:
```diff
@@ -184,9 +183,123 @@ def sample(self, n) -> None:
+
+        return labeled_examples
+
+    def get_chat_context(
```
docstring

rimashahbazyan replied:
done

ekmb commented on lm_eval/tasks/robustness/README.md:
```diff
@@ -0,0 +1,76 @@
+# ROBUSTNESS
```
header

ekmb commented on lm_eval/tasks/robustness/README.md:
```diff
@@ -0,0 +1,76 @@
+
+### Option order robustness
+
+Measures the model's robustness towards the placment of the correct answer in the options list by swapping the correct answer with all the other possible options.
```
typo: "...to the placement"

rimashahbazyan replied:
fixed

ekmb commented on lm_eval/tasks/robustness/README.md:
```diff
@@ -0,0 +1,76 @@
+
+Measures the model's robustness towards the same 10 prompts as the former task in a fewshot manner. This task is only available for MMLU-PRO.
+
+For evaluating `robustness_mmlu_pro_fewshot` it is required to pass the `"--fewshot_as_multiturn"` flag as well as set the `"--num_fewshot 5"`,
```
is there a way to set default for the overall robustness task?

rimashahbazyan replied:
No actually in their current implementation `--fewshot_as_multiturn` parameter could only be passed by argparser, there is no way to add this into yaml without changing some of the logic in the their evaluator class.
What concerns `--num_fewshot`, even though we cam pass it through yaml, but they have a assertion check where enforce passing `num_fewshot`>0 when `fewshot_as_multiturn` is True, but in this assertion they are looking to the parameter passed through argparser not the one in the yaml.

TLDR: there is not a way for setting this default for the overall robustness task without changes in their evaluator class.

ekmb commented on lm_eval/tasks/robustness/README.md:
```diff
@@ -0,0 +1,76 @@
+# ROBUSTNESS
+
+## Paper
```
Let's not mention non existent paper but references to mmlu-pro and agieval are needed

rimashahbazyan replied:
fixed

ekmb commented on lm_eval/tasks/robustness/agi_eval/option_format_robustness_agieval.yaml:
```diff
@@ -0,0 +1,37 @@
+  - option_format_robustness_agieval_sat_math
+
+aggregate_metric_list:
+  - metric: prompt_0_macro_accuracy
```
header +
what are these prompt_* here?
Does this one run eval on the default prompt?

ekmb replied:
are these format options? if so, - why are they called prompt*?

rimashahbazyan replied:
In terms of implementation format options and prompt robustness are the same task, with only the list of prompts being different, so I will remove the word "prompt" having only {index}_macro_accuracy, for the sake of avoiding confusion

rimashahbazyan committed:
+ License header added
+ docstrings added for sampler
- prompt_ removed from option fromat and prompt robustness tasks metric names
+ references added fro mmlu-pro and Agieval

rimashahbazyan committed:
Merge branch 'robustness_task' of github.com:rimashahbazyan/lm-evaluation-harness into robustness_task
